api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -583,26 +583,23 @@ public Integer getCountOfUsers(String name, List<Role> roles, boolean includeRet
 		if (name != null) {
 			name = StringUtils.replace(name,", ", " ");
 		}
-		
 		if (roles == null) {
-			roles = new ArrayList<>();
-		}
-		
-		// add the requested roles and all child roles for consideration
-		Set<Role> allRoles = new HashSet<>();
```

Comment:
you need to move the following code

```java
Set<Role> allRoles = new HashSet<>();
for (Role r : roles) {
...
}
```

right below the 

```java
Role authRole = getRole(RoleConstants.AUTHENTICATED);
if (roles.contains(authRole)) {
...
}
```

this is because we do not necessarily need to compute `allRoles` since as you can see there is a return statement in the `if` block and when this branch is reached we do not need `allRoles`.